optaplanner-core/src/main/java/org/optaplanner/core/config/solver/DomainAccessType.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.optaplanner.core.config.solver;
```

Comment:
1) Should this be in package "config", "api" or "impl"?
I'd argue "api", because I have a sneaking feeling we might see this in annotation parameters in the future too (see SolutionCloner parameters in @PlanningSolution).

1) package "solver" is not appropriate, we have the "domain" package

Reply:
2) package "solver" is not appropriate, we have the "domain" package which fits better (it doesn't matter that in config there is no domain package yet: if a subpackage exists in one of the 3 (api/config/impl), it exists in all three).

3) Is this worthy to be directly in the domain package or does it need a subpackage? I 'd argue it's not important enough to be at the top. How about "domain.common", "domain.lang" or "domain.access"?

Proposal A) org.optaplanner.core.api.domain.common

Let's meet up when @rsynek is available to decide on these naming discussions.

Reply:
Proposal A is decided in meeting.